examples/datasource-streaming-backend-websocket/streaming-backend-websocket-plugin/pkg/plugin/datasource.go

```diff
@@ -154,38 +93,65 @@ func (d *Datasource) CheckHealth(_ context.Context, req *backend.CheckHealthRequ
 	}, nil
 }
 
-func (d *Datasource) SubscribeStream(context.Context, *backend.SubscribeStreamRequest) (*backend.SubscribeStreamResponse, error) {
+// SubscribeStream just returns an ok in this case, since we will always allow the user to successfully connect.
+// Permissions verifications could be done here. Check backend.StreamHandler docs for more details.
+func (d *Datasource) SubscribeStream(_ context.Context, req *backend.SubscribeStreamRequest) (*backend.SubscribeStreamResponse, error) {
 	return &backend.SubscribeStreamResponse{
 		Status: backend.SubscribeStreamStatusOK,
 	}, nil
 }
 
+// PublishStream just returns permission denied in this case, since in this example we don't want the user to send stream data.
+// Permissions verifications could be done here. Check backend.StreamHandler docs for more details.
 func (d *Datasource) PublishStream(context.Context, *backend.PublishStreamRequest) (*backend.PublishStreamResponse, error) {
 	return &backend.PublishStreamResponse{
 		Status: backend.PublishStreamStatusPermissionDenied,
 	}, nil
 }
 
 func (d *Datasource) RunStream(ctx context.Context, req *backend.RunStreamRequest, sender *backend.StreamSender) error {
-	messages := d.wsClient.Read()
+	q := Query{}
+	json.Unmarshal(req.Data, &q)
+
+	limitsData, err := json.Marshal(q)
+	if err != nil {
+		return err
+	}
+
+	err = d.wsClient.Connect()
+	if err != nil {
+		return err
+	}
+	defer d.wsClient.Close()
+
+	err = d.wsClient.SendMessage(string(limitsData))
+	if err != nil {
+		return err
+	}
+
+	messages := d.wsClient.Read(ctx)
 
 	for {
 		select {
 		case <-ctx.Done():
-			return nil
+			return ctx.Err()
 		case rawMsg := <-messages:
 			var msg Message
-			json.Unmarshal([]byte(rawMsg), &msg)
+			if err := json.Unmarshal([]byte(rawMsg), &msg); err != nil {
+				Logger.Error("Failed to unmarshal message", "error", err)
```

Comment:
Here it should skip execution of the body, so you should change this to:

```suggestion
				Logger.Error("Failed to unmarshal message", "error", err)
				break	
```